cmd/skiff/layers.go

```diff
@@ -3,60 +3,87 @@ package main
 import (
 	"context"
 	"fmt"
+	"io"
+	"text/tabwriter"
 
 	"github.com/containers/image/v5/types"
+	"github.com/opencontainers/go-digest"
 	"github.com/urfave/cli/v3"
 
 	skiff "github.com/dcermak/skiff/pkg"
 )
 
-func ShowLayerUsage(ctx context.Context, sysCtx *types.SystemContext, uri string) (string, error) {
+func ShowLayerUsage(ctx context.Context, sysCtx *types.SystemContext, uri string, output io.Writer, fullDigest bool) error {
 	img, layers, err := skiff.ImageAndLayersFromURI(ctx, sysCtx, uri)
 	if err != nil {
-		return "", err
+		return err
 	}
 
 	inspect, err := img.Inspect(ctx)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	res := ""
-	if layers != nil && len(layers) > 0 {
+	w := tabwriter.NewWriter(output, 0, 8, 2, ' ', 0)
+	defer w.Flush()
+
+	if len(layers) > 0 {
 		if len(inspect.LayersData) != len(layers) {
-			return "", fmt.Errorf(
+			return fmt.Errorf(
 				"internal error: image inspect returned %d layers, storage returned %d layers",
 				len(inspect.LayersData),
 				len(layers),
 			)
 		}
-		for i, l := range inspect.LayersData {
-			res += fmt.Sprintf("%s %d %s %d\n", l.Digest, l.Size, layers[i].ID, layers[i].UncompressedSize)
+		fmt.Fprintln(w, "Diff ID\tUncompressed Size")
+		for _, l := range layers {
+			fmt.Fprintf(w, "%s\t%d\n", skiff.FormatDigest(l.UncompressedDigest, fullDigest), l.UncompressedSize)
 		}
 
 	} else {
-		for _, l := range inspect.LayersData {
-			res += fmt.Sprintf("%s %d\n", l.Digest, l.Size)
+		// in theory, the OCI Config contains the 'rootfs.diffids' array
+		// with the diffIDs i.e. the uncompressed digests
+		// => use that if available otherwise we fall back to compressed digests
+		conf, err := img.OCIConfig(ctx)
+		diffIDs := []digest.Digest{}
+
+		// only get them if the rootfs type is correct
+		if err == nil && conf.RootFS.Type == "layers" {
```

Comment:
```suggestion
		if err == nil && && conf != nil && conf.RootFS.Type == "layers" {
```

Reply:
not sure if `conf` can be `nil` if `err` is `nil`, but it won't hurt